packages/docusaurus-plugin-content-docs/src/sidebars/__tests__/generator.test.ts

```diff
@@ -473,4 +477,116 @@ describe('DefaultSidebarItemsGenerator', () => {
       },
     ] as Sidebar);
   });
+
+  test('respects custom isCategoryIndex', async () => {
+    const sidebarSlice = await DefaultSidebarItemsGenerator({
+      numberPrefixParser: DefaultNumberPrefixParser,
+      isCategoryIndex({fileName, directories}) {
+        return (
+          fileName.replace(
```

Comment:
🤯 why not using something simpler like file.endsWith("-index")

It was not so easy for me to understand the intent of this code 🤪

Reply:
The intention is to only match those with the same prefix as the category name, as described in #6255